rbi/core/enumerable.rbi

```diff
@@ -1562,6 +1562,16 @@ module Enumerable
   sig {returns(T::Enumerator[Elem])}
   def take_while(&blk); end
 
+  # Tallies the collection, i.e., counts the occurrences of each element.
+  # Returns a hash with the elements of the collection as keys and the
+  # corresponding counts as values.
+  #
+  # ```ruby
+  # ["a", "b", "c", "b"].tally  #=> {"a"=>1, "b"=>2, "c"=>1}
+  # ```
+  sig {returns(T::Hash[T.nilable(Elem), Integer])}
```

Comment:
It doesn't make sense to me why the key of the resulting hash would ever be `nil`.

If there were `nil`s in the enumerable, then of course the `Elem` type would include `nil`. But it doesn't make sense to me to allow `xs.tally[nil]`.

```suggestion
  sig {returns(T::Hash[Elem, Integer])}
```